### ISSUE TITLE
Training accuracy heavily impacted when using activation checkpointing

### ISSUE THREAD

**Issue by Moldoteck** [microsoft/DeepSpeed#1116]
I just use deepspeed without any additional params and let's get consistently 0.250 accuracy on 200'th batch. When I turn on the activation checkpointing, I get 0.025 accuracy with the same parameters at the same batch number.
My model is like an unet and has conv2d, leakyrelu and batchnorm layers and skip connections. Can this be a problem?
For skip connections I have modified a bit example from you and saved/consumed them when needed in a separate list

**Comment by Moldoteck** [comment]
Ok, after some investigation, i have found out a workaround>
https://github.com/microsoft/DeepSpeed/discussions/1117#discussioncomment-820377
But I don't get why it is expecting to perform that _requires_grad_()_ operation
